### PR TITLE
Update Anomaly Detection API

### PR DESCRIPTION
**Pull request by juaduan**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific langauge SDKs and CLIs that must be refreshed after the swagger is published.
      - [x] SDK of .NET (need service team to ensure code readiness)
      - [x] SDK of Python
      - [x] SDK of Java
      - [x] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that label “WaitForARMFeedback” will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits.

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by heaths** [comment]
Could you please rebase your changes atop main to avoid showing so many unrelated commits requiring review from many different people. It's also difficult to tell which of the 712 files you need reviewed.

Assuming you're on your topic branch (note: best practice is to always create a branch - not work directly in main, even if pushing to your fork), upstream is Azure/azure-rest-api-specs, and origin is juaduan/azure-rest-api-specs:

```bash
git fetch upstream main
git rebase upstream/main
#resolve any merge conflicts
git push -u --force origin main
```

This will make the PR much cleaner to review.

**Comment by markweitzel** [comment]
@juaduan -- We've been experimenting with the [Spectral Linter](https://aka.ms/azapi/style) which has a nice VSCode plugin. We've used this to update have a quick look at the OpenAPI doc and apply the recently updated [REST API Guidelines](https://aka.ms/azapi/guidelines). We've identified a few things that would be good to look at and/or talk through...

There are a number of guidelines spread throughout your document that are easy to address, for example:
* Your Error response should contain a `x-ms-error-code` header.
* All success responses except 202 & 204 should define a response body.
* Property name should be camel case.

**Comment by juaduan** [comment]
@markweitzel thanks for your recommendation. We are walking through the REST API guidelines to modify our documents.

**Review by markweitzel — CHANGES_REQUESTED** [review]
Based on the Stewardship board meeting, some ideas are captured in the issue.

**Comment by juaduan** [comment]
Hi @markweitzel, we have updated our PR based on your suggestions. Please take a review. Thank you.

**Review by mikekistler — COMMENTED** [review]
I left a few quetions/comments I hope you can address.

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.1-preview.1/AnomalyDetector.json [hunk 1]
```diff
@@ -0,0 +1,587 @@
...
+        "linear",
+        "fixed",
+        "zero",
+        "notFill"
+      ]
+    },
+    "ImputeFixedValue": {
+      "type": "number",
+      "format": "float",
+      "x-nullable": false,
```
[the remark this answers is not in the document]
Comment (moreOver0) [reply specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.1-preview.1/AnomalyDetector.json]: Removed all the `x-nullable: false`

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.1-preview.1/AnomalyDetector.json [hunk 2]
```diff
@@ -0,0 +1,587 @@
...
+      "description": "Custom Interval is used to set non-standard time interval, for example, if the series is 5 minutes, request can be set as {\"granularity\":\"minutely\", \"customInterval\":5}."
+    },
+    "ImputeMode": {
+      "type": "string",
+      "description": "Define the impute method, can be one of auto, previous, linear, fixed, zero, notFill.",
+      "x-nullable": false,
+      "x-ms-enum": {
+        "name": "ImputeMode",
+        "modelAsString": false,
```
[the remark this answers is not in the document]
Comment (moreOver0) [reply specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.1-preview.1/AnomalyDetector.json]: Good point. Do make this `modelAsString` as `true`.

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.1-preview.1/AnomalyDetector.json [hunk 3]
```diff
@@ -0,0 +1,587 @@
...
+    "CustomInterval": {
+      "type": "integer",
+      "format": "int32",
+      "x-nullable": false,
+      "description": "Custom Interval is used to set non-standard time interval, for example, if the series is 5 minutes, request can be set as {\"granularity\":\"minutely\", \"customInterval\":5}."
+    },
+    "ImputeMode": {
+      "type": "string",
+      "description": "Define the impute method, can be one of auto, previous, linear, fixed, zero, notFill.",
+      "x-nullable": false,
```
[the remark this answers is not in the document]
Comment (moreOver0) [reply specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.1-preview.1/AnomalyDetector.json]: Removed all the `x-nullable: false`

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.1-preview.1/AnomalyDetector.json [hunk 4]
```diff
@@ -0,0 +1,587 @@
...
+        "message": {
+          "description": "A message explaining the error reported by the service.",
+          "type": "string"
+        }
+      }
+    },
+    "TimeGranularity": {
+      "type": "string",
+      "description": "Optional argument, can be one of yearly, monthly, weekly, daily, hourly, minutely, secondly, microsecond or none. If granularity is not present, it will be none by default. If granularity is none, the timestamp property in time series point can be absent.",
+      "x-nullable": true,
```
[the remark this answers is not in the document]
Comment (moreOver0) [reply specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.1-preview.1/AnomalyDetector.json]: Removed the `x-nullable`. I double confirmed with the team, this was be modified mistakenly.

**Comment by moreOver0** [comment]
@mikekistler Thanks for review again, we've addressed the comments.

**Comment by mengaims** [comment]
@mikekistler Thanks for reviewing! We've commit a change other than resolving the comments: added "severity" to required fields of EntireDetectResponse and LastDetectResponse. Sorry for the late change, could you help confirm whether it's valid?

**Comment by mengaims** [comment]
@mikekistler Thanks for the advices! We have completed your suggested change about "xs-ms-error-code" and adding descriptions, could you please check and help sign off?

**Review by mikekistler — APPROVED** [review]
Looks good. 👍 

I left a few minor comments you should consider.

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.1-preview.1/MultivariateAnomalyDetector.json [hunk 5]
```diff
@@ -0,0 +1,1236 @@
...
+            "headers": {
+              "Location": {
+                "type": "string",
+                "description": "Location and ID of the model."
+              }
+            }
+          },
+          "default": {
+            "description": "Error response",
+            "x-ms-error-response": true,
```
[the remark this answers is not in the document]
Comment (juaduan) [reply specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.1-preview.1/MultivariateAnomalyDetector.json]: Hi @mikekistler, thank you for your comments. We have modified the swagger file per your suggestion.

`x-ms-error-response` has been removed in the default response.

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.1-preview.1/MultivariateAnomalyDetector.json [hunk 6]
```diff
@@ -0,0 +1,1236 @@
...
+  },
+  "schemes": [
+    "https"
+  ],
+  "paths": {
+    "/multivariate/models": {
+      "post": {
+        "responses": {
+          "201": {
+            "description": "Submit a multivariate model training task successfully.",
```
[the remark this answers is not in the document]
Comment (juaduan) [reply specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.1-preview.1/MultivariateAnomalyDetector.json]: The description message has been changed to "Create and train a multivariate anomaly detection model.".

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.1-preview.1/MultivariateAnomalyDetector.json [hunk 7]
```diff
@@ -0,0 +1,1236 @@
...
+        ],
+        "responses": {
+          "200": {
+            "description": "Export the multivariate model as a zip package.",
+            "headers": {
+              "content-type": {
+                "description": "application/zip",
+                "type": "string"
+              }
+            },
```
[the remark this answers is not in the document]
Comment (juaduan) [reply specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.1-preview.1/MultivariateAnomalyDetector.json]: This section is removed and the content-type has been specified in `produces`.

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.1-preview.1/MultivariateAnomalyDetector.json [hunk 8]
```diff
@@ -0,0 +1,1236 @@
...
+            "schema": {
+              "type": "object",
+              "format": "file"
```
[the remark this answers is not in the document]
Comment (juaduan) [reply specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.1-preview.1/MultivariateAnomalyDetector.json]: The type and format has been modified per your suggestion.

**Review by jhendrixMSFT — APPROVED** [review]

**Comment by jhendrixMSFT** [comment]
@markweitzel have your changes been addressed?  Your review is still "changes requested"

**Pull request merged**